xbmc/platform/darwin/tvos/TVOSTopShelf.mm

```diff
@@ -62,14 +62,10 @@
     CLog::Log(LOGDEBUG, "[TopShelf] (sandboxed: {}) Use storeUrl: {}", isSandboxed ? "yes" : "no",
               storeUrl.path.UTF8String);
 
-    // store all old thumbs in array
     auto fileManager = NSFileManager.defaultManager;
-    auto filePaths =
-        [NSMutableSet setWithArray:[fileManager contentsOfDirectoryAtPath:storeUrl.path error:nil]];
-    std::string raPath = storeUrl.path.UTF8String;
 
-    // Shared dicts  (if we are sandboxed we use sharedDefaults, else we use sharedJailbreak)
-    auto sharedDefaults =
+    // Shared dicts (if we are sandboxed we use sharedSandbox, else we use sharedJailbreak)
+    auto sharedSandbox =
         isSandboxed ? [[NSUserDefaults alloc] initWithSuiteName:[tvosShared getSharedID]] : nil;
     const auto sharedJailbreakUrl = [storeUrl URLByAppendingPathComponent:@"shared.dict"];
     NSMutableDictionary* sharedJailbreak = nil;
```

Comment:
Is there really a need to have a sharedSandbox and a sharedJailbreak? They are both just dicts that are added/removed, with from what i can see the only difference being what happens in the end. Why not just use the one dict, and do a isSandboxed check for the synchronize/writeurl at the end?

Reply:
`sharedSandbox` is a `NSUserDefaults` and `sharedJailbreak` is a `NSMutableDictionary`.
If I want to work with only one variable I need to:

1. If sandboxed --> Copy existing NSUserDefaults values in a NSMutableDictionary var. Else if jailbroken --> Open a NSMutableDictionary var from the existing `shared.dict` file
2. Update the NSMutableDictionary var according to the TopShelf category changes
3. If sandboxed --> clear all existing NSUserDefaults values and copy every key/value from the NSMutableDictionary var to the NSUserDefaults, then synchronize. Else if jailbroken --> Write the NSMutableDictionary var in `shared.dict` file.

Do I have good? If yes, in sandboxed case, the "ping pong" between NSUserDefaults and NSMutableDictionary seems heavy? No?

Reply:
After some test I can confirm that `[[NSUserDefaults alloc] initWithSuiteName:[tvosShared getSharedID]]` doesn't work when Kodi is fake-signed. We need to keep two separate dict :-/